pallets/otc-settlements/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+		if otc.partially_fillable {
+			let price_diff = {
+				if otc_price > router_price_after {
+					otc_price.saturating_sub(router_price_after)
+				} else {
+					router_price_after.saturating_sub(otc_price)
+				}
+			};
+
+			let price_precision = T::PricePrecision::get()
+				.checked_mul(&otc_price)
+				.ok_or(ArithmeticError::Overflow)?;
+			if price_diff > price_precision {
+				ensure!(router_price_after <= otc_price, Error::<T>::TradeAmountTooHigh);
+				ensure!(router_price_after >= otc_price, Error::<T>::TradeAmountTooLow);
+			}
+		}
+
+		let asset_a_balance_after_router_trade = <T as Config>::Currency::free_balance(asset_a, &pallet_acc);
+
+		let profit = asset_a_balance_after_router_trade
+			.checked_sub(amount)
+			.ok_or(ArithmeticError::Overflow)?;
+
+		Self::ensure_min_profit(otc.amount_in, profit)?;
+
+		<T as Config>::Currency::transfer(asset_a, &pallet_acc, &T::ProfitReceiver::get(), profit)?;
+
+		<T as Config>::Currency::withdraw(asset_a, &pallet_acc, amount)?;
+
+		let asset_a_balance_after = <T as Config>::Currency::free_balance(asset_a, &pallet_acc);
+		let asset_b_balance_after = <T as Config>::Currency::free_balance(asset_b, &pallet_acc);
+
+		ensure!(
+			asset_a_balance_after == asset_a_balance_before,
+			Error::<T>::BalanceInconsistency
+		);
+		ensure!(
+			asset_b_balance_after == asset_b_balance_before,
+			Error::<T>::BalanceInconsistency
+		);
+
+		Self::deposit_event(Event::Executed {
+			asset_id: asset_a,
+			profit,
+		});
+
+		Ok(())
+	}
+
+	/// Store the latest block number in the offchain storage.
+	/// Returns `true` if `block_number` is newer than the block number stored in the storage.
+	fn try_update_last_block_storage(block_number: BlockNumberFor<T>) -> bool {
+		let last_update_storage = StorageValueRef::persistent(OFFCHAIN_WORKER_DATA_LAST_UPDATE);
+		let last_update = last_update_storage
+			.get::<BlockNumberFor<T>>()
+			.unwrap_or_default()
+			.unwrap_or_default();
+
+		if block_number > last_update {
+			last_update_storage.set(&block_number);
+			true
+		} else {
+			false
+		}
+	}
+
+	/// Sort open OTCs orders and save a list in the offchain storage.
+	fn sort_otcs(block_number: BlockNumberFor<T>) {
+		log::debug!(
+			target: "offchain_worker::sort_otcs",
+			"sort_otcs()");
+
+		// acquire offchain worker lock.
+		let lock_expiration = Duration::from_millis(LOCK_TIMEOUT_EXPIRATION);
+		let mut lock = StorageLock::<'_, Time>::with_deadline(SORTED_ORDERS_LOCK, lock_expiration);
+
+		if Self::try_update_last_block_storage(block_number) {
+			if let Ok(_guard) = lock.try_lock() {
+				let sorted_otcs = StorageValueRef::persistent(OFFCHAIN_WORKER_DATA);
+
+				let mut list = vec![];
+				for (otc_id, otc) in <pallet_otc::Orders<T>>::iter() {
+					let otc_price = FixedU128::checked_from_rational(otc.amount_out, otc.amount_in);
+
+					let route = T::Router::get_route(AssetPair {
+						asset_in: otc.asset_out,
+						asset_out: otc.asset_in,
```

Comment:
this is confusing. i would suggest to add a comment why out is in and in is out.